### PR TITLE
vlinsert: added opentelemetry logs support

### LINKED ISSUE #4839: Send logs from otel-collector to victorialogs
### Is your question request related to a specific component?

 When I try to send logs to victorialogs using the loki exporter in otel-collector, I encounter the following error. What could be the cause of this?
 
### Opentelemetry-collector logs:
```
error   exporterhelper/queued_retry.go:401      Exporting failed. The error is not retryable. Dropping data.   
       {"kind": "exporter", "data_type": "logs", "name": "loki", "error": 
       "Permanent error: HTTP 400 \"Bad Request\": remoteAddr: \"10.244.1.81:50446\"; 
       requestURI: /insert/loki/api/v1/push; unsupported path requested: \"/insert/loki/api/v1/push\"", "dropped_items": 80}
```
### victorialogs logs:
```
 {"ts":"2023-08-15T06:48:45.376Z","level":"warn","caller":"VictoriaMetrics/lib/httpserver/httpserver.go:348","msg":"remoteAddr: 
       \"10.244.1.81:51858\"; requestURI: /insert/loki/api/v1/push; unsupported path requested: \"/insert/loki/api/v1/push\""}
```

```
curl -H "Content-Type: application/json" -XPOST "http://10.1.133.153:9428/insert/loki/api/v1/push?_stream_fields=instance,job" --data-raw \
  '{"streams": [{ "stream": { "instance": "host123", "job": "app42" }, "values": [ [ "0", "foo fizzbuzz bar" ] ] }]}'
```
```
error:  
unsupported path requested: "/insert/loki/api/v1/push"
```

>otel-collector configmap
```
apiVersion: v1
data:
  relay: |
    exporters:
      prometheus:
        endpoint: 0.0.0.0:9464
      clickhouse:
        endpoint: tcp://10.1.242.16:9000
        username: admin
        password: 27ff0399-0d3a-4bd8-919d-17c2181e6fb9
        database: otel
        ttl_days: 3
        logs_table_name: otel_logs
        traces_table_name: otel_traces
        metrics_table_name: otel_metrics
        timeout: 5s
        retry_on_failure:
          enabled: true
          initial_interval: 5s
          max_interval: 30s
          max_elapsed_time: 300s
      prometheusremotewrite:
        endpoint: http://10.1.190.177:8428/api/v1/write
      loki:
        endpoint: "http://10.1.133.153:9428/insert/loki/api/v1/push"
    extensions:
      health_check: {}
      memory_ballast:
        size_in_percentage: 40
    processors:
      batch:
        timeout: 5s
        send_batch_size: 100000
    receivers:
      otlp:
        protocols:
          grpc:
            endpoint: ${MY_POD_IP}:4317
          http:
            cors:
              allowed_origins:
              - http://*
              - https://*
            endpoint: ${MY_POD_IP}:4318
      prometheus:
        config:
          scrape_configs:
            - job_name: "kubernetes-metricsresource"
              # Default to scraping over https. If required, just disable this or change to
              # `http`.


```

### Troubleshooting docs

- [ ] General - https://docs.victoriametrics.com/Troubleshooting.html
- [ ] vmagent - https://docs.victoriametrics.com/vmagent.html#troubleshooting
- [ ] vmalert - https://docs.victoriametrics.com/vmalert.html#troubleshooting

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
### Describe Your Changes

Fixes #4839

Added support of OpenTelemetry logs for Victoria Logs

### Checklist

The following checks are **mandatory**:

- [ ] My change adheres [VictoriaMetrics contributing guidelines](https://docs.victoriametrics.com/contributing/).

### CODE REVIEW

**Comment by gergelyurbancsik** [comment]
Hi @ArtemVoitsekhovskyi @AndrewChubatiuk AndrewChubatiuk
What is the plan for releasing this ?

**Comment by AndrewChubatiuk** [comment]
hey @gergelyurbancsik
better to ask @valyala 😉

**Review by f41gh7 — CHANGES_REQUESTED** [review]
As discussed with @AndrewChubatiuk following changes required:

1) make vlinsert/openetelemetry similar to the other ingestion components:
 * add CanWrite function
 *. split json and pb ingestion path into separate functions
2) remove logstorage dependency from lib/protoparser

File: app/vlinsert/opentelemetry/pb.go [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+	SeverityText         string
+	Body                 *pb.AnyValue
+	Attributes           []*pb.KeyValue
+}
+
+func (r *LogRecord) severity() string {
+	if r.SeverityText != "" {
+		return r.SeverityText
+	}
+	return logSeverities[r.SeverityNumber]
```
Comment (f41gh7) [inline app/vlinsert/opentelemetry/pb.go:211]: Could it panic?

Comment (AndrewChubatiuk) [reply app/vlinsert/opentelemetry/pb.go:211]: it should not, it covers all subset of allowed values defined in spec, but i can also account it

**Review by f41gh7 — APPROVED** [review]
LGTM

**Pull request merged**